test/main.go

```diff
@@ -0,0 +1,76 @@
+package main
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"text/template"
+)
+
+const templateFile = "../dir-index.html"
+
+// Copied from go-ipfs/core/corehttp/gateway_indexPage.go
+type listingTemplateData struct {
+	Listing  []directoryItem
+	Path     string
+	BackLink string
+	Hash     string
+}
+
+type directoryItem struct {
+	Size string
+	Name string
+	Path string
+}
+
+var testData = listingTemplateData{
+	Listing: []directoryItem{{
+		Size: "25 MiB",
+		Name: "short-film.mov",
+		Path: "short-film.mov",
+	}, {
+		Size: "1 KiB",
+		Name: "description.txt",
```

Comment:
```suggestion
		Name: "this-piece-of-paper's-got-47-words-37-sentences-58-words-we-wanna-know.txt",
```
For troubleshooting long-name errors.